databuilder/query_model_transforms.py

```diff
@@ -68,10 +69,34 @@ def add_selected_columns_to_pick_row(nodes):
         force_setattr(node, "selected_columns", selected_columns)
 
 
+def include_all_selected_columns_in_sorts(nodes):
+    """
+    We want to ensure that results are consistent when picking the first or last of a sorted
+    frame, even when the sorting specified in the variable definition isn't sufficient to completely
+    specify the order. To that end we add extra sorts for each column that is ultimately going to
+    be returned from the sorted table.
+
+    This has the additional benefit that the order is the same between different DBMSes, which
+    doesn't help our users but makes testing a bit easier.
+    """
+    for node in nodes:
+        if not isinstance(node, PickOneRowPerPatientWithColumns):
```

Comment:
Why would this ever be hit?

Reply:
Each transformation inspects every single node.